src/integration/resources/coordinator_client.go

```diff
@@ -881,3 +882,80 @@ func toResponse(
 
 	return nil
 }
+
+// GraphiteQuery retrieves graphite raw data.
+func (c *CoordinatorClient) GraphiteQuery(
+	graphiteReq GraphiteQueryRequest,
+) ([]Datapoint, error) {
+	if graphiteReq.From.IsZero() {
+		graphiteReq.From = time.Now().Add(-24 * time.Hour)
+	}
+	if graphiteReq.Until.IsZero() {
+		graphiteReq.Until = time.Now()
+	}
+
+	queryStr := fmt.Sprintf(
+		"%s?target=%s&from=%d&until=%d",
+		graphite.ReadURL, graphiteReq.Target,
+		graphiteReq.From.Unix(),
+		graphiteReq.Until.Unix(),
+	)
+
+	url := c.makeURL(queryStr)
+	logger := c.logger.With(
+		ZapMethod("graphiteQuery"), zap.String("url", url))
+	logger.Info("running")
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		logger.Error("failed get", zap.Error(err))
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if status := resp.StatusCode; status != http.StatusOK {
+		return nil, fmt.Errorf("query response status not OK, received %v", status)
```

Comment:
Can we also include the body in this error message so the client has the resaon for the non-200 response.